packages/perspective-viewer-d3fc/src/js/series/seriesColors.js

```diff
@@ -26,40 +26,69 @@ export function seriesColorsFromGroups(settings) {
             }
         }
     });
-    return fromDomain(domain);
+    return colorScale().domain(domain)();
 }
 
-function fromDomain(domain) {
-    return domain.length > 1 ? d3.scaleOrdinal(d3.schemeCategory10.map(withOpacity)).domain(domain) : null;
-}
+export function colorScale() {
+    let domain = null;
+    let defaultColours = null;
+    let mapFunction = withOpacity;
 
-export function withoutOpacity(color) {
-    const lastComma = color.lastIndexOf(",");
-    const valueIndex = color.indexOf("(");
-    return valueIndex !== -1 && lastComma !== -1 ? `rgb(${color.substring(valueIndex + 1, lastComma)})` : color;
-}
+    const colors = () => {
+        if (defaultColours || domain.length > 1) {
+            const range = domain.length > 1 ? d3.schemeCategory10.map(mapFunction) : defaultColours;
```

Comment:
Is `defaultColours` only supposed to be used if we have a domain of 1 or less?

Reply:
Yes, and it's only needed for canvas charts, and just avoids having a fallback in the chart code to style the colour if the colourSeries is `null`. For Svg charts, we fall back on styling with css instead.

Reply:
👍